### DISCUSSION TITLE
vue-cli-serivce won't add new components to component library

### DISCUSSION THREAD

**Discussion by arishojaei** [vuejs/vue-cli#6801] [Q&A]
I've created a project with `vue-cli` and added a few components but after I run this below script and publish dist to npm, I just can use 2 component in my nuxt project.

```
vue-cli-service build src/index.ts --target lib --name zarin --modern && rm ./dist/demo.html
```

```ts
// src/index.ts
// This file used to build component by vue-cli-service

import { VueConstructor } from 'vue';
import * as components from './components';

const install = (Vue:VueConstructor) => {
  Object.entries(components).forEach(([name, component]) => {
    Vue.component(name, component);
  });
};

if (typeof window !== 'undefined' && window.Vue) {
  install(window.Vue);
}

export default install;
export * from './components';
```

**I'm not sure if I build my component right or wrong because I get many `umd` files!**

![Screen Shot 2021-11-02 at 12 54 01 PM](https://user-images.githubusercontent.com/4179565/139820273-9f9482f6-d979-4608-9a71-3f29a9aa3561.png)

**Comment by fangbinwei** [comment]
Could you provide a  minimal reproduction?

**Reply by arishojaei** [reply]
I'm not sure to understand what exactly you need

**Reply by fangbinwei** [reply]
A repo that can reproduce your problem

**Reply by arishojaei** [reply]
I can't do that, sorry
